eva/binder/statement_binder.py

```diff
@@ -151,7 +152,36 @@ def _bind_delete_statement(self, node: DeleteTableStatement):
     @bind.register(CreateMaterializedViewStatement)
     def _bind_create_mat_statement(self, node: CreateMaterializedViewStatement):
         self.bind(node.query)
-        # Todo Verify if the number projected columns matches table
+        num_projected_columns = 0
+        # TODO we should fix the binder. All binded object should have the same interface.
+        for expr in node.query.target_list:
+            if expr.etype == ExpressionType.TUPLE_VALUE:
+                num_projected_columns += 1
+            elif expr.etype == ExpressionType.FUNCTION_EXPRESSION:
+                num_projected_columns += len(expr.output_objs)
+            else:
+                raise BinderError("Unsupported expression type {}.".format(expr.etype))
+
+        assert (
+            len(node.col_list) == 0 or len(node.col_list) == num_projected_columns
+        ), "Projected columns mismatch, expected {} found {}.".format(len(node.col_list), num_projected_columns)
+        binded_col_list = []
+        idx = 0
+        for expr in node.query.target_list:
+            output_objs = [(expr.col_name, expr.col_object)] \
+                if expr.etype == ExpressionType.TUPLE_VALUE \
+                else zip(expr.projection_columns, expr.output_objs)
+            for col_name, output_obj in output_objs:
+                binded_col_list.append(
+                    ColumnDefinition(
```

Comment:
Why is the bound column a ColumnDefinition? There is nothing to bind here. We should change the variable name to avoid confusion. Thanks!